gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -383,7 +715,7 @@ gst_tensor_src_iio_get_property (GObject * object, guint prop_id,
       break;
 
     case PROP_FREQUENCY:
-      g_value_set_uint64 (value, self->sampling_frequency);
+      g_value_set_ulong (value, self->sampling_frequency);
```

Comment:
As Linux kernel uses unsigned-long for sampling frequency, we need to be ready for different bit-width for this property as you are currently doing.

However, please make sure that the results (srcpad) of this plugin consistent regardless of architecture. The pipeline itself may be distributed over different computers. (e.g., tensor_iio_src in 32bit system, it's user tensor_filter in 64bit system)

Reply:
I'd recommend to use 64bit internal representation and accept both 32/64 representations for inputs from Linux IIO.

Reply:
Updated.
Internally treated with uint64. Exposure to outside access is constrained with ulong.